### DISCUSSION TITLE
Documentation on materials

### DISCUSSION THREAD

**Discussion by vegaonline** [MPh-py/MPh#46] [General]
I think documentation on materials needs a bit elaborate write-up. 
(a) If an mph file exists with global material(s) and/or material links to domains, then documentation is required to describe information to access those materials including updation of property like conductivity, permittivity etc.
(b) The reverse is also required. If one wants to create material database taking data from global materials database and then wishes to link with different domains, how one may do with this MPh code.

**Comment by john-hen** [comment]
I understand how this would be nice to have. But it would also extend the scope of this project quite considerably.

For specific questions like this one, the MPh documentation, in a number of places, currently refers the reader to Comsol's [Programming Reference Manual][japi] for the Java API. The MPh documentation does explain [how to translate Java code to Python code][jstyle] and also [how to make such code more "pythonic"][pstyle]. But that reference manual has a total of 970 pages, and I, for one, have neither the time nor the inclination to cover all of that information in the MPh docs only so that the code examples are in Python instead of Java.

I'm not saying we cannot document specifics like that. If you want to contribute that, you are very welcome to do so. The documentation is structured roughly along the guidelines laid out by the [Diátaxis Framework][diataxis]. What you suggest would be a "How-to guide", which corresponds to what I called the ["Demonstrations"][demos] chapter. So we could neatly add a guide there on how to manage the definitions of materials.


[japi]: https://comsol.com/documentation/COMSOL_ProgrammingReferenceManual.pdf
[jstyle]: https://mph.readthedocs.io/en/1.0/demonstrations.html#creating-models-java-style
[pstyle]: https://mph.readthedocs.io/en/1.0/demonstrations.html#creating-models-python-style
[diataxis]: https://diataxis.fr
[demos]: https://mph.readthedocs.io/en/1.0/demonstrations.html

**Reply by vegaonline** [reply]
I am trying to use MPh to simulate a transmon qubit with simple RF modal simulation to get eigenfrequencies. I shall push it here making a branch with a readme where I shall show how to select material and link domains so that the issues I faced could be cleared to other users. 

This project must stay.

**Reply by john-hen** [reply]
Okay, great.

**Reply by Trophime** [reply]
@vegaonline Any update?
